qlib/TableMapper.qm

```diff
@@ -841,18 +919,53 @@ on_error table_mapper.rollback();
             remove rowcode;
         }
 
-        #! flushes queued data to the database
-        private hash flushIntern() {
+        /**
+         * flushes queued data to the database
+         * @param force_flush to flush even if there is not enough data to fill a block,
+         *                    e.g. on explicit flush() call
+         * @return Qore::NOTHING when nothing was flushed, otherwise returns
+         *         a bulk (hash of lists) with the data that were flushed into the DB,
+         *         updated by the 'ret_args' from the DB.
+         */
+
+        # return nothing if nothing needs to be flushed
+        private *hash flushIntern(bool force_flush) {
+            if (!exists hbuf) {
+                return; # nothing to flush
+            }
+            *int rec_list_size = getRecListSize(hbuf);
+            if (exists rec_list_size) {
```

Comment:
what about something simpler:
```
# return if there is no data or we we haven't reached the buffer size and force_flush is not set
if (!rec_list_size || (rec_list_size < insert_block && !force_flush))
    return;
```
for me the above 2 lines are much easier to read than 14 lines of if else logic